### ISSUE TITLE
Robust address-checking

### ISSUE THREAD

**Issue by ch4ot1c** [BTCPrivate/bitcoin-private-full-node-wallet#20]
Currently, the prefixes are checked, but I think the gui relies on the full-node to handle invalid addresses. This should be processed ahead-of-time for best UX

**Comment by Pega88** [comment]
would a snippet like 

`    public boolean isAddressValid(String address) {
        if (address.startsWith("b") || address.startsWith("zk")) {
            try {
                Base58.decodeToBigInteger(address);
                Base58.decodeChecked(address);
            } catch (AddressFormatException e) {
                e.printStackTrace();
                return false;
            }
            return true;
        }else{
            return false;
        }
    }`

suffice?

**Comment by ch4ot1c** [comment]
Yes - here is where the Base58 import briefly happens (with bitcoinj)

https://github.com/BTCPrivate/bitcoin-private-full-node-wallet/blob/41fbfba19e47a90d53dfb7eda95959dc231b5a5c/src/main/java/org/btcprivate/wallets/fullnode/util/Util.java#L325
